mlflow/entities/trace_info.py

```diff
@@ -46,15 +48,15 @@ def to_proto(self):
         for key, value in self.request_metadata.items():
             attr = ProtoTraceRequestMetadata()
             attr.key = key
-            attr.value = value
+            attr.value = str(value)[:MAX_CHARS_IN_TRACE_INFO_METADATA_AND_TAGS]
```

Comment:
Doing the validation at the end as tags/request_metadata can be set to `TraceInfo` in any place within MLflow code, not necessarily via client.